tests/testsuite/support/mod.rs

```diff
@@ -399,6 +399,27 @@ impl Project {
         execs
     }
 
+    /// Safely run a process after `cargo build`.
+    ///
+    /// Windows has a problem where a process cannot be reliably
+    /// be replaced, removed, or renamed immediately after executing it.
+    /// The action may fail (with errors like Access is denied), or
+    /// it may succeed, but future attempts to use the same filename
+    /// will fail with "Already Exists".
+    ///
+    /// If you have a test that needs to do `cargo run` multiple
+    /// times, you should instead use `cargo build` and use this
+    /// method to run the executable. Each time you call this,
+    /// use a new name for `dst`.
+    /// See https://github.com/rust-lang/cargo/issues/5481
+    pub fn safely_rename_run(&self, src: &str, dst: &str) -> Execs {
```

Comment:
👨‍🎨 maybe just called `run` and with a name automatically derived from src?

Reply:
Yea, it's a bit wordy. However, I don't want to infer that this is a normal way of running an executable.  I dropped the "safely" bit.

To derive the name, it would need to maintain state somewhere to keep the name unique.  `Project` isn't generally mutable, so it seemed a bit awkward. So I'm unsure if there's a clean way to do that.

Reply:
> Yea, it's a bit wordy. However, I don't want to infer that this is a normal way of running an executable. I dropped the "safely" bit.

It avoids a pitfall on Windows, under certain circumstances.  If there's no penalty hit I wouldn't mind people using it as the normal way of running an executable.  WDYT?

> To derive the name, it would need to maintain state somewhere to keep the name unique. `Project` isn't generally mutable, so it seemed a bit awkward. So I'm unsure if there's a clean way to do that.

I'd just set up another one of these as a run naming counter:

https://github.com/rust-lang/cargo/blob/87dd6c279ab8b499118677733958e29362793d6e/tests/testsuite/support/paths.rs#L12-L14

Reply:
I would prefer if normal execution goes through `cargo run`. It's a somewhat different codepath which has had some issues in the past, so I think it's good for it to be fully exercised.  `cargo run` is also a bit more convenient to use.

Reply:
Ah, I see.